action/protocol/staking/protocol.go

```diff
@@ -206,15 +209,47 @@ func (p *Protocol) CreatePreStates(ctx context.Context, sm protocol.StateManager
 	bcCtx := protocol.MustGetBlockchainCtx(ctx)
 	blkCtx := protocol.MustGetBlockCtx(ctx)
 	hu := config.NewHeightUpgrade(&bcCtx.Genesis)
-	if blkCtx.BlockHeight != hu.GreenlandBlockHeight() {
+	if blkCtx.BlockHeight == hu.GreenlandBlockHeight() {
+		csr, err := ConstructBaseView(sm)
+		if err != nil {
+			return err
+		}
+		if _, err = sm.PutState(csr.BaseView().bucketPool.total, protocol.NamespaceOption(StakingNameSpace), protocol.KeyOption(bucketPoolAddrKey)); err != nil {
+			return err
+		}
+	}
+
+	if p.candBucketsIndexer == nil {
+		return nil
+	}
+	rp := rolldpos.MustGetProtocol(protocol.MustGetRegistry(ctx))
+	epochStartHeight := rp.GetEpochHeight(rp.GetEpochNum(blkCtx.BlockHeight))
+	if epochStartHeight != blkCtx.BlockHeight {
 		return nil
 	}
-	csr, err := ConstructBaseView(sm)
+
```

Comment:
i think in previous logic, if we put data every block, then the data of an epoch is the last block data, now the data of an epoch is the first block data. not sure if that matter

Reply:
I think the frist block recorded the last block's data those are settled 
the candidates and buckets data are not settled yet in the first block

Reply:
for example: epoch 0 is block 1~720
handleStakingIndexer() should be called at time of block 720, but not at block 1